sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -65,45 +89,44 @@ bool oclHandleInvalidWorkGroupSize(const device_impl &DeviceImpl,
           PI_INVALID_WORK_GROUP_SIZE);
   }
-
-  if (Ver[0] == '1') {
-    // OpenCL 1.x:
-    // PI_INVALID_WORK_GROUP_SIZE if local_work_size is specified and the
-    // total number of work-items in the work-group computed as
-    // local_work_size[0] * ... * local_work_size[work_dim – 1] is greater
-    // than the value specified by PI_DEVICE_MAX_WORK_GROUP_SIZE in
-    // table 4.3
-    size_t MaxWGSize = 0;
-    Plugin.call<PiApiKind::piDeviceGetInfo>(
-        Device, PI_DEVICE_INFO_MAX_WORK_GROUP_SIZE, sizeof(size_t), &MaxWGSize,
-        nullptr);
-    const size_t TotalNumberOfWIs =
-        NDRDesc.LocalSize[0] * NDRDesc.LocalSize[1] * NDRDesc.LocalSize[2];
-    if (TotalNumberOfWIs > MaxWGSize)
-      throw sycl::nd_range_error(
-          "Total number of work-items in a work-group cannot exceed "
-          "info::device::max_work_group_size which is equal to " +
-              std::to_string(MaxWGSize),
-          PI_INVALID_WORK_GROUP_SIZE);
-  } else {
-    // OpenCL 2.x:
-    // PI_INVALID_WORK_GROUP_SIZE if local_work_size is specified and the
-    // total number of work-items in the work-group computed as
-    // local_work_size[0] * ... * local_work_size[work_dim – 1] is greater
-    // than the value specified by PI_KERNEL_GROUP_INFO_WORK_GROUP_SIZE in
-    // table 5.21.
-    size_t KernelWGSize = 0;
-    Plugin.call<PiApiKind::piKernelGetGroupInfo>(
-        Kernel, Device, PI_KERNEL_GROUP_INFO_WORK_GROUP_SIZE, sizeof(size_t),
-        &KernelWGSize, nullptr);
-    const size_t TotalNumberOfWIs =
-        NDRDesc.LocalSize[0] * NDRDesc.LocalSize[1] * NDRDesc.LocalSize[2];
-    if (TotalNumberOfWIs > KernelWGSize)
-      throw sycl::nd_range_error(
-          "Total number of work-items in a work-group cannot exceed "
-          "info::kernel_work_group::work_group_size which is equal to " +
-              std::to_string(KernelWGSize) + " for this kernel",
-          PI_INVALID_WORK_GROUP_SIZE);
+  if (Platform.get_backend() == cl::sycl::backend::opencl) {
+    if (Ver[0] == '1') {
+      // OpenCL 1.x:
+      // PI_INVALID_WORK_GROUP_SIZE if local_work_size is specified and the
+      // total number of work-items in the work-group computed as
+      // local_work_size[0] * ... * local_work_size[work_dim – 1] is greater
+      // than the value specified by PI_DEVICE_MAX_WORK_GROUP_SIZE in
+      // table 4.3
+      size_t MaxWGSize = 0;
+      Plugin.call<PiApiKind::piDeviceGetInfo>(
+          Device, PI_DEVICE_INFO_MAX_WORK_GROUP_SIZE, sizeof(size_t),
+          &MaxWGSize, nullptr);
+      const size_t TotalNumberOfWIs =
+          NDRDesc.LocalSize[0] * NDRDesc.LocalSize[1] * NDRDesc.LocalSize[2];
+      if (TotalNumberOfWIs > MaxWGSize)
+        throw sycl::nd_range_error(
+            "Total number of work-items in a work-group cannot exceed " +
+                std::to_string(MaxWGSize),
+            PI_INVALID_WORK_GROUP_SIZE);
+    } else {
+      // RELEVENT // OpenCL 2.x:
```

Comment:
I see you added the word "RELEVANT" to this comment.  Was that a note to yourself indicating that this arm of the `if` statement was relevant to L0 (non-OpenCL backends)?  If so, I think you should either change the code like this:

```
if (Platform.get_backend() == cl::sycl::backend::opencl && OpenClVer[0] == '1') {
  // Do the first check
}
else {
  // Do the second check
}
```

Or, if you are planning to do this as part of a subsequent PR, add a more informative TODO comment indicating what needs to be done.